flamingpy/codes/graphs/egraph.py

```diff
@@ -214,3 +217,132 @@ def draw_adj(self, **kwargs):
 
         adj = self.adj_generator(sparse=False)
         return plot_mat_heat_map(adj, **kwargs)
+
+    def add_qubit(
+        self,
+        qubit: Union[None, tuple] = None,
+        neighbors: Union[None, list] = None,
+        macro: Union[None, tuple] = None,
+    ) -> None:
+        """Add qubit to EGraph connected to neighbors.
+
+        Args:
+            qubit (tuple[int, int, int] or None): qubit to add. If a 3-tuple,
+                the qubit is added in that position. If qubit is None, the qubit is
+                positioned one unit further than the maximum position in the z
+                direction, in position (x_max + 1, 0, 0).
+
+            neighbours (list[int], list[tuple], or None): neighbors of qubit specified
+                with indices or positions.
+
+            macro (tuple[int, int, int], optional): the macronode into which to add
+                the qubit. The EGraph must be macronized.
```

Comment:
Sorry, just realized something: the way we treat qubits inside macronodes is that their position (e.g. (0.9, 0, 0)) should round to the tuple corresponding to the macronode (e.g. (1, 0, 0)). Therefore, let us change the `macro` option to a boolean `add_to_macro` with the description:

```suggestion
            add_to_macronode (bool, optional): if True, add qubit to a macronode.
                The qubit must be a tuple that rounds to an integer tuple corresponding
                to the macronode.
```